payments/payments.go

```diff
@@ -0,0 +1,41 @@
+package payments
+
+import (
+	"context"
+	"net/http"
+)
+
+const (
+	StripeProvider = "stripe"
+	PayPalProvider = "paypal"
+)
+
+// Provider represents a payment provider that can optionally charge, refund,
+// preauthorize payments.
+type Provider interface {
```

Comment:
I am not in love with doing an interface for each of them. I think that if we did 
```
type Provider interface {
  Name () string
  NewTransactionHandler(ctx, r) (TransactionHandler, error)
}

type TransactionHandler interface {
Charge...
Refund...
Preauthorize...
}
```

it would be simpler. You pretty much have that right now.